### PR TITLE
Feat/new map users markers

### LINKED ISSUE #4265: Create Users' Markers Controllers
**Acceptance Criteria**
- [x] Create a `Hot Region Controller` that is up to decide which users belong to the hot region and which do not
- [x] Create a `User Icon` prefab and corresponding pools according to the usage
- [x] Create a `Hot Users Markers Controller`
   - [x] With Frequency X updates the position of the markers in the hot region
- [x] Create a `Cold Users Markers Controller`
   - [x] With Frequency Y (where Y < X) updates the position of the markers in the cold region
- [x] Updates do not happen if the region is culled

### PR DESCRIPTION
**Pull request by mikhail-dcl**
## What does this PR change?

Closes #4265

- Hot Users Markers
- Cold Users Markers
- Unity Object Pool to wrap instantiation nicely
- HotSceneController rework to fit the `async` scheme 
- Tests

Does not include:
- Constructed prefabs
- Integration

Will be doing in on the final stage

...

## How to test the changes?

1. Go to: https://play.decentraland.zone/?explorer-branch={branch_name}&{desired_url_params}
2. ...

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by AjimenezDCL — APPROVED** [review]
Looks good, left a few comments!

File: unity-renderer/Assets/DCLServices/MapRendererV2/MapLayers/UsersMarkers/ColdArea/UsersMarkersColdAreaController.cs [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            int drawOrder);
+
+        internal const int CREATE_PER_BATCH = 20;
+        internal const int COMMS_RADIUS_THRESHOLD = 2;
+
+        private Service<IHotScenesFetcher> hotScenesFetcher;
+
+        private readonly ColdUserMarkerObject prefab;
+        private readonly int maxMarkers;
+        private readonly ColdUserMarkerBuilder builder;
```
[the remark this answers is not in the document]
Comment (mikhail-dcl) [reply unity-renderer/Assets/DCLServices/MapRendererV2/MapLayers/UsersMarkers/ColdArea/UsersMarkersColdAreaController.cs:32]: It's not quite a pool, check `ColdUserMarkersStorage`

File: unity-renderer/Assets/DCLServices/MapRendererV2/MapLayers/UsersMarkers/ColdArea/UsersMarkersColdAreaController.cs [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        protected override void DisposeImpl()
+        {
+            realmName.OnChange -= OnRealmNameChange;
+            userPosition.OnChange -= OnUserPositionChange;
+            storage.Dispose();
+        }
+
+        private async UniTaskVoid ColdAreasUpdateLoop(CancellationToken cancellationToken)
+        {
+            await foreach (var data in hotScenesFetcher.Ref.ScenesInfo)
```
[the remark this answers is not in the document]
Comment (mikhail-dcl) [reply unity-renderer/Assets/DCLServices/MapRendererV2/MapLayers/UsersMarkers/ColdArea/UsersMarkersColdAreaController.cs:118]: Did it

File: unity-renderer/Assets/DCLServices/MapRendererV2/MapLayers/UsersMarkers/ColdArea/UsersMarkersColdAreaController.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (mikhail-dcl) [reply unity-renderer/Assets/DCLServices/MapRendererV2/MapLayers/UsersMarkers/ColdArea/UsersMarkersColdAreaController.cs:32]: We don't need anything but the `Func<T>` in my case so pool would be excessive

[the remark this answers is not in the document]
Comment (mikhail-dcl) [reply unity-renderer/Assets/DCLServices/MapRendererV2/MapLayers/UsersMarkers/ColdArea/UsersMarkersColdAreaController.cs:32]: Also there is a `List` or `LinkedList` behind the pool capabilities of which I don't need here

**Pull request merged**